docker/kong/Dockerfile

```diff
@@ -1,4 +1,9 @@
 FROM kong:2.6.0-alpine
+
+# Deploy the plugin and dependencies for local testing
 USER root
-RUN luarocks install lua-resty-openssl
-USER kong
+COPY ./plugin/*.lua          /tmp/oauth-proxy/
+COPY ./plugin/kong*.rockspec /tmp/oauth-proxy/
+RUN cd /tmp/oauth-proxy/ && luarocks make kong-oauth-proxy-1.0.0-1.rockspec
```

Comment:
If I remember correctly Docker caches individual instructions to optimize builds. I wonder if `luarocks make` will be run again if any of the files copied in the instructions above changed.

Reply:
According to the note at the end of the COPY documentation: https://docs.docker.com/engine/reference/builder/#copy the cache for run commands is invalidated whenever copied files changed, so we should be fine here.